pulsar-client/src/main/java/org/apache/pulsar/client/impl/PatternTopicsConsumerImpl.java

```diff
@@ -36,11 +50,119 @@ public PatternTopicsConsumerImpl(Pattern topicsPattern,
                               CompletableFuture<Consumer> subscribeFuture) {
         super(client, conf, listenerExecutor, subscribeFuture);
         this.topicsPattern = topicsPattern;
+
+        if (this.namespaceName == null) {
+            this.namespaceName = getNameSpaceFromPattern(topicsPattern);
+        }
+        checkArgument(getNameSpaceFromPattern(topicsPattern).toString().equals(this.namespaceName.toString()));
+
+        this.topicsChangeListener = new PatternTopicsChangedListener();
+        scheduledExecutor = (ScheduledExecutorService)listenerExecutor;
```

Comment:
It's probably better to not use the listener executor, because its purpose is to be blocked by the application. 

When that happens, there is no guarantee that the consumer will discover the new topics within any give timeframe.

Reply:
Thanks, will change back to each consumer have one executor.

Reply:
No no :) 

We should reuse a different executor from the `ClientImpl` instance.

Reply:
For example, we could use `timer` in `ClientImpl`